### PR TITLE
Task[56337][56419] TabBar + Create UI Home [Deploy] Deploy 16/11/2022

### PR DESCRIPTION
**Pull request by DoQuangKhanh0209**
### [Related Tickets]
- Task #56337: [TabBar]
- Task #56419: Create UI Home Screen
## [WHAT]
Create TabBar with 4 items (Home,Upcomming, Search, Dowload)
 - Go to the destination view
 - Selected Tab and Unselected Tab
Creating a user interface for the Home screen
 - Logo icon
 - Bulging background hot movies
 - List trending movies using TableView
- List trending TV movies using TableView
- List popular using TableView
- List upcoming movies using TableView
- List top raced using TableView
- Button Movies
- Title label
- Button person
- Button play 
- Button Info
- Button My List"
#### Evidence
simulator_screenshot_FE260CB1-10B1-4D8D-A5B4-9EE3D14ECBA7

https://user-images.githubusercontent.com/114377716/202116973-c1f78786-6311-43fd-8bcb-8cfc164a8707.MP4

### CODE REVIEW

File: Netflix Movies/Netflix Movies/Controllers/DownloadsViewController.swift [hunk 1]
```diff
@@ -0,0 +1,17 @@
+//
+//  DownloadsViewController.swift
+//  Netflix Movies
+//
+//  Created by Khanh on 15/11/2022.
+//
+
+import UIKit
+
+class DownloadsViewController: UIViewController {
```
Comment (namtn-2711) [inline Netflix Movies/Netflix Movies/Controllers/DownloadsViewController.swift]: final

File: Netflix Movies/Netflix Movies/Controllers/HomeViewController.swift [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+//  Netflix Movies
+//
+//  Created by Khanh on 15/11/2022.
+//
+
+import UIKit
+
+final class HomeViewController: UIViewController {
+    
+    let sectionTitles: [String] = ["Trending Movies", "Popular", "Trending TV", "Upcomming", "Top Reced"]
```
Comment (namtn-2711) [inline Netflix Movies/Netflix Movies/Controllers/HomeViewController.swift:12]: dùng enum

File: Netflix Movies/Netflix Movies/Controllers/HomeViewController.swift [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+//            UIBarButtonItem(image: UIImage(named: "namelogo")?.withRenderingMode(.alwaysOriginal), style: .done, target: self, action: nil)
+//        ]
+
+        navigationItem.rightBarButtonItems = [
+            UIBarButtonItem(image: UIImage(systemName: "person"), style: .done, target: self, action: nil),
+            UIBarButtonItem(image: UIImage(systemName: "play.rectangle"), style: .done, target: self, action: nil)
+        ]
+        navigationController?.navigationBar.tintColor = .black
+    }
+    //set scroll
```
Comment (namtn-2711) [inline Netflix Movies/Netflix Movies/Controllers/HomeViewController.swift]: remove cmt

File: Netflix Movies/Netflix Movies/Controllers/SearchViewController.swift [hunk 4]
```diff
@@ -0,0 +1,16 @@
+//
+//  SearchViewController.swift
+//  Netflix Movies
+//
+//  Created by Khanh on 15/11/2022.
+//
+
+import UIKit
+
+class SearchViewController: UIViewController {
```
Comment (namtn-2711) [inline Netflix Movies/Netflix Movies/Controllers/SearchViewController.swift]: final

File: Netflix Movies/Netflix Movies/Controllers/HomeViewController.swift [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        header.textLabel?.textColor = .black
+    }
+        
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return 200
+    }
+        
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return 40
+    }
```
Comment (namtn-2711) [inline Netflix Movies/Netflix Movies/Controllers/HomeViewController.swift]: tạo num layout

File: Netflix Movies/Netflix Movies/Views/CollectionViewTableViewCell.swift [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+extension CollectionViewTableViewCell: UICollectionViewDelegate, UICollectionViewDataSource {
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "cell", for: indexPath)
+        cell.backgroundColor = .green
+        return cell
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        10
```
Comment (namtn-2711) [inline Netflix Movies/Netflix Movies/Views/CollectionViewTableViewCell.swift]: không hard code

File: Netflix Movies/Netflix Movies/Views/HeroHeaderUIVIew.swift [hunk 7]
```diff
@@ -0,0 +1,103 @@
+//
+//  HeroHeaderUIVIew.swift
+//  Netflix Movies
+//
+//  Created by Quang Khánh on 16/11/2022.
+//
+
+import UIKit
+
+class HeroHeaderUIVIew: UIView {
```
Comment (namtn-2711) [inline Netflix Movies/Netflix Movies/Views/HeroHeaderUIVIew.swift]: final + Dùng file xib cho gọn e

File: Netflix Movies/Netflix Movies/Controllers/HomeViewController.swift (hunk 2 above)
Comment (DoQuangKhanh0209) [reply Netflix Movies/Netflix Movies/Controllers/HomeViewController.swift:12]: a ơi đây là data e tạo tạm thôi chưa có API mai e mới tạo Networking

File: Netflix Movies/Netflix Movies/MainTabBarViewController.swift [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+        viewDownload.title = "Downloads"
+        
+        viewHome.tabBarItem.image = UIImage(systemName: "house")
+        viewUpcoming.tabBarItem.image = UIImage(systemName: "play.circle")
+        viewSearch.tabBarItem.image = UIImage(systemName: "magnifyingglass")
+        viewDownload.tabBarItem.image = UIImage(systemName: "arrow.down.to.line")
+        
+        tabBar.tintColor = .label
+        
+        setViewControllers([viewHome, viewUpcoming,viewSearch ,viewDownload], animated: true)
```
Comment (thuannv-2723) [inline Netflix Movies/Netflix Movies/MainTabBarViewController.swift]: Tạo ra 1 hàm setupTabBar hoặc configTabBar

**Pull request merged**